clang/lib/AST/ASTContext.cpp

```diff
@@ -2791,6 +2791,12 @@ bool ASTContext::hasUniqueObjectRepresentations(
     return hasUniqueObjectRepresentations(getBaseElementType(Ty),
                                           CheckIfTriviallyCopyable);
 
+  if (Ty->isVoidType())
```

Comment:
It is not clear why we need this change as well.

Reply:
For the new `!Ty->isIncompleteType()` assertion below. Before `void` would go to the `!isTriviallyCopyable` path, this is just a more explicit way of handling `__has_unique_object_representations(void)`. The assertion could also be `assert(Ty->isVoidType() || !Ty->isIncompleteType())` to follow the spec better?

Reply:
I'd prefer that, I think